src/internal.c

```diff
@@ -12730,6 +12752,7 @@ int DoApplicationData(WOLFSSL* ssl, byte* input, word32* inOutIdx)
     dataSz = msgSz - ivExtra - ssl->keys.padSz;
     if (dataSz < 0) {
         WOLFSSL_MSG("App data buffer error, malicious input?");
+        SendAlert(ssl, alert_fatal, unexpected_message);
```

Comment:
How is a bad size on App Data an unexpected message?

Reply:
I believe the interpretation is that if the length is all wrong then the data wan't encrypted and that this is an unexpected message - expecting an encrypted message.